PSKoans/Koans/Constructs and Patterns/AboutRegularExpressions.Koans.ps1

```diff
@@ -0,0 +1,334 @@
+        #>
+
+        # The portion of this regex in curly braces acts as a quantifier for the symbol that comes immediately before it. It means "exactly 4 digits."
+        'I can count to 1024' -match '\d{4}' | should -BeTrue
+        $matches[0] | should -Be '__'
+
+        '__' -match '\w{10}' | should -BeTrue
+        '__' -match '\w{10}' | should -BeFalse
+
+        # One can use curly braces to give a range of numbers. This one means "between 2 and 4 of any character"
+        'Paper airplane' -match '.{2,4}'
+
+        $fruitMatch = 'p'
+        $fruitMatch += '__'
+        [regex]::Matches('Apples', $fruitMatch).Value | Should -Be 'pp'
+        $fruitMatch.Contains('}') | should -BeTrue     # Make sure you use curly braces!
+
+        # If you omit the second number, the custom quantifier becomes "this many or more". This one means "two or more non-whitespace characters"
+        [bool]$whitespaceMatch = __ # Should be either $true or $false
+        'The moon in June is a big balloon' -match '\s{2,}' | should -Be $whitespaceMatch
+    }
+
+    # round brackets
+    it 'groups patterns together' {
+        <#
+            Round brackets are used to create groups of symbols. These groups can then have quantifiers applied to the entire group, or be used in
+            approximately 42 trillion other ways. This is just the tip of the round bracket/regular expression groups iceberg.
+        #>
+
+        [regex]::Matches('Bears Beat Bongos', '(B.+){3}').Value | should -Be '__'
+
+        $groupingPattern = '__'
+        [regex]::Matches('BadgerBadgerBadger', $groupingPattern).Value | should -Be 'BadgerBadgerBadger'
+        $groupingPattern.Contains(')') | should -BeTrue    # Use a group!
+
+    }
+
+    # square brackets
+    it 'defines a set' {
+        <#
+            Square brackets denote a set, or array/collection, of symbols within a regular expression. Imagine a pattern that might read "a or b
+            or c or d". You may use square brackets to create that regex. The set contained within the square brackets represent one character
+            within the string being searched.
+        #>
+
+        [bool]$eolMatch = __ # Should be either $true or $false
+        'End of the line' -match '[efg]$' | should -Be $eolMatch
+
+        '__' -match '[abcdefg]$' | should -BeTrue
+
+        # Normally, the ^ character means "start of a line", but inside of square brackets, it negates the set. Instead of "these symbols" it means "not these symbols"
+        '__' -match '[^abcdefg]$' | should -BeTrue
+
+        # What do you think you might do if you want to match a literal ^ symbol within a set? Use the ^ (caret) symbol in this one.
+        $caretMatch = '[__]'
+        '^&*' -match $caretMatch | should -BeTrue
+        $caretMatch.Contains('^') | should -BeTrue      # Use the caret symbol!
+    }
+}
+
+describe 'Medatative Examples' {
```

Comment:
```suggestion
Describe 'Meditative Examples' {
```
Pop these in a file in `Katas` and number it `350`.. and then add some really evil puzzles 😁

Reply:
🤔 though I guess it's alright to have a few transitional puzzles here at the end.

Reply:
More puzzles in a katas... coming... one day :D